### ISSUE TITLE
{Linked:Bug10274492}After enabling High Contrast, unable to see the selection when highlighting text: A11y_MonacoEditor_ No Disruption of Accessibility Features

### ISSUE THREAD

**Issue by AnKushSingh05** [microsoft/monaco-editor#4945]
### GitHubTags:
#A11yTCS; #A11ySev2; #A11yMAS; #WCAG4.3.1; #Desktopweb; #Win11; #Visual Studio Code Client; #VisualStudioCode-Web-Apr2021; #ChromiumEdge; #a11y-monacoeditor; #DevTools; #PartnerBug; # No Disruption of Accessibility Features:#ED:Bug10274492;

### Environment Details:
#URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#XQAAAAJjAQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscwzZypW_1uhs0-3YGTBOsVJ4S4FwfCDQb5-2uUHXTVAefbVSd9IzaZ_vv9VR9KugsKimARHSWlrTPOQHROITHxmAetfeRrm3VHdSe7_Etv6WpgzlH4o4eQ6dMnzwfwYcUqOiZ_ge3TaAC-S-3X8NnmdsBgoPipQD3aPDiLZZPK6l7xCFp8ZECZbOE6TnVWRQ5d9hpoWU6fPQxCmvXOHG0yQjnpBbUFcTVnmLIgKYp_MMpyC98ktGgml-3rke-U2SAGxZ_LvB8DSxLE1KkD9hE5voTWZUnA5LHB1fNAbw8DB8v_jVO8Ng
Application name: Monaco Editor
Chromium Edge: Version 138.0.3351.95 (Official build) (64-bit)
Windows 11 Enterprise (Version 23H2)  
OS Build 22631.3737

### Repro Steps:

1. Hit the URL: https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#example-creating-the-editor-hello-world.
2. Monaco editor page will appear.
3. Apply high contrast.
4. Observe that In After enabling High Contrast, unable to see the selection when highlighting text.

### Actual Results:
After enabling High Contrast, unable to see the selection when highlighting text.

### Expected Results:
After applying High Contrast, User should be able to see the selection while selecting text..

### User Impact:
Low vision users may have trouble identifying selected text if they cannot see the selection when using High Contrast mode.

### Attachment:

https://github.com/user-attachments/assets/72a5db4d-b958-4691-9bda-62a3747c0117

**Comment by meganrogge** [comment]
How are you enabling that theme?

**Comment by meganrogge** [comment]
They replied they're setting this on windows:

<img width="1315" height="497" alt="Image" src="https://github.com/user-attachments/assets/91ce5f8c-94e7-44a6-abfd-c379759c6e1f" />

**Comment by AnKushSingh05** [comment]
Hi @meganrogge I am still able to repro the issue can you please reopen this.

**Comment by meganrogge** [comment]
@AnKushSingh05 are you on the latest build?

**Comment by hawkticehurst** [comment]
Hi @AnKushSingh05 we were able to validate the fix is working on our end. 

Can you make sure that you have switched to the latest developer build? We haven't shipped this change to stable yet, so you need to be testing on the developer build. To do this, open the dropdown in the top right corner of the page and switch from `latest stable (0.53.0)` to `0.54.0-dev-20250924`.

As a final note, you should expect to see that the Windows high contrast themes are no longer applied to the editor. The change we made was to disable forced color adjustment for the Monaco editor. This change matches what we do in VS Code desktop app and now the editor will rely on the custom [built in API / support for setting themes](https://microsoft.github.io/monaco-editor/docs.html#interfaces/editor.IGlobalEditorOptions.html#theme) only –– which out the box ships with VS Code's high contrast themes.

<img width="1474" height="625" alt="Image" src="https://github.com/user-attachments/assets/bd35a259-bb34-4d43-84d2-6436f1b4d26a" />

**Comment by AnKushSingh05** [comment]
> Hi [@AnKushSingh05](https://github.com/AnKushSingh05) we were able to validate the fix is working on our end.
> 
> Can you make sure that you have switched to the latest developer build? We haven't shipped this change to stable yet, so you need to be testing on the developer build. To do this, open the dropdown in the top right corner of the page and switch from `latest stable (0.53.0)` to `0.54.0-dev-20250924`.
> 
> As a final note, you should expect to see that the Windows high contrast themes are no longer applied to the editor. The change we made was to disable forced color adjustment for the Monaco editor. This change matches what we do in VS Code desktop app and now the editor will rely on the custom [built in API / support for setting themes](https://microsoft.github.io/monaco-editor/docs.html#interfaces/editor.IGlobalEditorOptions.html#theme) only –– which out the box ships with VS Code's high contrast themes.
> 
> <img alt="Image" width="1474" height="625" src="https://private-user-images.githubusercontent.com/39639992/493533661-bd35a259-bb34-4d43-84d2-6436f1b4d26a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTg3NzgyNDgsIm5iZiI6MTc1ODc3Nzk0OCwicGF0aCI6Ii8zOTYzOTk5Mi80OTM1MzM2NjEtYmQzNWEyNTktYmIzNC00ZDQzLTg0ZDItNjQzNmYxYjRkMjZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI1VDA1MjU0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5ZGYyODRhYTM5NWU0MzgxYWEwOTkxZGRhNDEyNzAxZjE5NGM4MDMwYTkxZTU0YzE5YmRiNmUxMzI0NGIxZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-Twwz3XUiR80rnR9IKjRZ6Egl5dcxh5c5GblHlRJz-c">

Just to confirm users will now need to manually set the high contrast theme using the Monaco editor's built-in API.
There’s no longer any automatic detection or application of system-level high contrast settings — the editor will rely solely on the explicitly set theme via the API.

**Comment by meganrogge** [comment]
Yes

**Comment by AnKushSingh05** [comment]
Issue is fixed in version 0.54.0-dev-20250924
https://microsoft.github.io/monaco-editor/playground.html?source=v0.54.0-dev-20250925#XQAAAAKgAQAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscwzvswRgmSaumWUF-TBOsVJ4S4FwfCDQb5-2uUHXTVAefbVSd9IzaZ_vv9VR9KugsKimARHSWlrTPOQHROITHxmAeFEs9_m3VHdSe7_Etv6WpgzlH4mWNgZDIl-yX5tau0K3KEjjNFWt5DeDSSh9uwCFVPl-wOx4YyfO3gjMW2X3QZWrOIN1nwoyJwVysPaJPxODXWDwlnY-5JMaTBnKdk2C-82rbBJiNzY7xoUflSVeEanuhhqDCk7sPi651sGfS4uv8iy4mPa1rDwX4mkie6tE3BWyzSjguX6cltvPf0FzgL0EfmvliWCRgkFMZyRSeZnrS0RMelKLwBBCdmzkNnausTLf1R8jw

<img width="1279" height="506" alt="Image" src="https://github.com/user-attachments/assets/ffdd6027-260a-4628-b58a-e65690f93214" />
